src/app/admin/activities/create-activity.form.tsx

```diff
@@ -0,0 +1,243 @@
+import { useForm } from "react-hook-form"
+import { toast } from "sonner"
+
+import { MarkdownEditor } from "@/components/md-editor"
+import { Button } from "@/components/ui/button"
+import { DateTimePicker } from "@/components/ui/date-time-picker"
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form"
+import { Input } from "@/components/ui/input"
+import {
+  type CreateActivityInput,
+  CreateActivityInputDto,
+} from "@/contracts/activity/activity"
+import { api } from "@/trpc/react"
+
+export type CreateActivityProps = {
+  close: () => void
+  refetch: () => void
+}
+
+export const CreateActivityForm: FC<CreateActivityProps> = ({
+  close,
+  refetch,
+}) => {
+  const createActivity = api.activity.create.useMutation({
+    onError: (error) => {
+      const title =
+        error instanceof Error ? error.message : "Activity create failed"
+      toast.error(title)
+      close()
+    },
+  })
+
+  const form = useForm<CreateActivityInput>({
+    resolver: zodResolver(CreateActivityInputDto),
+    defaultValues: {
+      title: "",
+      imageBase64: "",
+      description: "",
+      stageName: "",
+      startTime: new Date().toISOString(),
+      endTime: new Date().toISOString(),
+      publicVisibility: false,
+      archived: false,
+      reason: "",
+    },
+  })
+
+  const [hasTime, setHasTime] = useState(true)
+
+  const onSubmit = useCallback(async (values: CreateActivityInput) => {
+    try {
+      await createActivity.mutateAsync(values)
+      toast.success("Activity created")
+      refetch()
+      close()
+    } catch (error) {
+      console.error("Error in createActivity.mutateAsync", error)
+    }
+  }, [])
+
+  return (
+    <Form {...form}>
+      <form onSubmit={form.handleSubmit(onSubmit)} className={"space-y-3"}>
+        <FormField
+          control={form.control}
+          name={"title"}
+          render={({ field }) => (
+            <FormItem>
+              <FormLabel>Title</FormLabel>
+              <FormControl>
+                <Input placeholder={"Title"} {...field} />
+              </FormControl>
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+        <FormField
+          control={form.control}
+          name="imageBase64"
+          render={({ field }) => (
+            <FormItem>
+              <FormItem>
+                <FormLabel>Image</FormLabel>
+                <FormMessage>
+                  {field.value && (
+                    <Image
+                      src={field.value}
+                      width="250"
+                      height="250"
+                      alt="Decorative"
+                    />
+                  )}
+                </FormMessage>
+                <FormControl>
+                  <Input
+                    type="file"
+                    onChange={(e) => {
+                      if (e.target.files && e.target.files.length > 0) {
```

Comment:
this can be converted into its own component when we require something similar again. I'll create an issue for it :)